### DISCUSSION TITLE
nextcloud-aio-talk commented "unhealthy" in portainer since last AiO update

### DISCUSSION THREAD

**Discussion by punkyard** [nextcloud/all-in-one#2463] [Questions] [answered]
### Host OS 
AiO Docker debian 11

#### Nextcloud AIO version 
v5.0.0

#### Other valuable info 
![Capture d’écran 2023-05-02 à 11 05 24](https://user-images.githubusercontent.com/59349105/235625694-45cdad8c-e001-4e4c-ac22-e639a86fedd3.png)

Portainer's Logs
`Crypto: OpenSSL >= 1.1.0
[WARN] The libsrtp installation does not support AES-GCM profiles
hub.go:200: Using a maximum of 8 concurrent backend connections per host
hub.go:207: Using a timeout of 10s for backend connections
hub.go:303: Not using GeoIP database
main.go:228: Could not initialize janus MCU (dial tcp 127.0.0.1:8188: connect: connection refused) will retry in 1s
No cert/key specified, autogenerating some...
0: (17): ERROR: CONFIG: Empty cli-password, and so telnet cli interface is disabled! Please set a non empty cli-password!
0: (17): INFO: Certificate file found: //turn_server_cert.pem
0: (17): INFO: Private key file found: //turn_server_pkey.pem
0: (17): WARNING: NO EXPLICIT LISTENER ADDRESS(ES) ARE CONFIGURED
0: (17): INFO: ===========Discovering listener addresses: =========
0: (17): INFO: Listener address to use: 127.0.0.1
0: (17): INFO: Listener address to use: 172.22.0.4
0: (17): INFO: =====================================================
0: (17): INFO: Total: 1 'real' addresses discovered
0: (17): INFO: =====================================================
0: (17): WARNING: NO EXPLICIT RELAY ADDRESS(ES) ARE CONFIGURED
0: (17): INFO: ===========Discovering relay addresses: =============
0: (17): INFO: Relay address to use: 172.22.0.4
0: (17): INFO: =====================================================
0: (17): INFO: Total: 1 relay addresses discovered
0: (17): INFO: =====================================================
Fingerprint of our certificate: F9:35:E6:02:AE:5A:B9:16:06:6B:EF:C4:05:CF:DC:87:F7:63:8E:AD:E1:6F:77:4E:19:75:EB:D0:2C:12:96:BF
Sessions watchdog started
Event handlers support disabled
Plugins folder: /usr/lib/janus/plugins
Loading plugin 'libjanus_recordplay.so'...
Joining Janus requests handler thread
JANUS Record&Play plugin initialized!
Loading plugin 'libjanus_echotest.so'...
JANUS EchoTest plugin initialized!
Loading plugin 'libjanus_duktape.so'...
[echotest.js]  Loading script...
[echotest.js]  Modules folder: /usr/share/janus/duktape
[echotest.js]  Loading module: janus-sdp
[echotest.js]  Module loaded
[echotest.js]  Script loaded
[echotest.js]  Initializing...
[echotest.js]  Initialized
Janus JavaScript plugin (Duktape) initialized!
Loading plugin 'libjanus_nosip.so'...
JANUS NoSIP plugin initialized!
Loading plugin 'libjanus_sip.so'...
JANUS SIP plugin initialized!
Loading plugin 'libjanus_lua.so'...
[echotest.lua] Loading...
[echotest.lua] Loaded
[echotest.lua] Initializing...
[echotest.lua] Initialized
Janus Lua plugin initialized!
Loading plugin 'libjanus_videocall.so'...
JANUS VideoCall plugin initialized!
Loading plugin 'libjanus_audiobridge.so'...
JANUS AudioBridge plugin initialized!
Loading plugin 'libjanus_videoroom.so'...
JANUS VideoRoom plugin initialized!
Loading plugin 'libjanus_streaming.so'...
JANUS Streaming plugin initialized!
Loading plugin 'libjanus_textroom.so'...
JANUS TextRoom plugin initialized!
Transport plugins folder: /usr/lib/janus/transports
Loading transport plugin 'libjanus_http.so'...
HTTP transport timer started
HTTP webserver started (port 8088, /janus path listener)...
JANUS REST (HTTP/HTTPS) transport plugin initialized!
Loading transport plugin 'libjanus_mqtt.so'...
MQTT SSL support disabled
[WARN] MQTT support disabled for both Janus and Admin API, giving up
JANUS MQTT transport plugin destroyed!
[WARN] The 'janus.transport.mqtt' plugin could not be initialized
Loading transport plugin 'libjanus_pfunix.so'...
[WARN] No Unix Sockets server started, giving up...
[WARN] The 'janus.transport.pfunix' plugin could not be initialized
Loading transport plugin 'libjanus_nanomsg.so'...
JANUS Nanomsg transport plugin initialized!
Nanomsg thread started
Loading transport plugin 'libjanus_websockets.so'...
libwebsockets logging: 0
Websockets server started (port 8188)...
JANUS WebSockets transport plugin initialized!
WebSockets thread started
Loading transport plugin 'libjanus_rabbitmq.so'...
RabbitMQ SSL support disabled
[WARN] RabbitMQ support disabled for both Janus and Admin API, giving up
[WARN] The 'janus.transport.rabbitmq' plugin could not be initialized
mcu_janus.go:294: Connected to Janus WebRTC Server 1.1.3 by Meetecho s.r.l.
mcu_janus.go:300: Found JANUS VideoRoom plugin 0.0.9 by Meetecho s.r.l.
mcu_janus.go:305: Data channels are supported
mcu_janus.go:307: WARNING: Full-Trickle is NOT enabled in Janus!
mcu_janus.go:311: Maximum bandwidth 1048576 bits/sec per publishing stream
mcu_janus.go:312: Maximum bandwidth 2097152 bits/sec per screensharing stream
Creating new session: 2447059625079982; 0x7f08a37005a0
mcu_janus.go:318: Created Janus session 2447059625079982
Creating new handle in session 2447059625079982: 274521831320082; 0x7f08a37005a0 0x7f08a4068630
mcu_janus.go:325: Created Janus handle 274521831320082
main.go:263: Using janus MCU
hub.go:385: Using a timeout of 10s for MCU requests
backend_server.go:105: No IPs configured for the stats endpoint, only allowing access from 127.0.0.1
main.go:339: Listening on 0.0.0.0:8081`

INSPECT 
`{
    "AppArmorProfile": "docker-default",
    "Args": [
        "/usr/bin/supervisord",
        "-c",
        "/supervisord.conf"
    ],
    "Config": {
        "AttachStderr": false,
        "AttachStdin": false,
        "AttachStdout": false,
        "Cmd": [
            "/usr/bin/supervisord",
            "-c",
            "/supervisord.conf"
        ],
        "Domainname": "",
        "Entrypoint": [
            "start.sh"
        ],
        "Env": [
            "NC_DOMAIN=my.domain.net",
            "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
            "SIGNALING_SECRET=1234",
            "TALK_PORT=3478",
            "TURN_SECRET=1234",
            "TZ=Europe/Paris"
        ],
        "ExposedPorts": {
            "3478/tcp": {},
            "3478/udp": {},
            "5349/tcp": {},
            "5349/udp": {}
        },
        "Healthcheck": {
            "Test": [
                "CMD-SHELL",
                "(nc -z localhost 8081 && nc -z localhost 8188 && nc -z localhost 4222 && nc -z localhost \"$TALK_PORT\" && nc -z \"$NC_DOMAIN\" \"$TALK_PORT\") || exit 1"
            ]
        },
        "Hostname": "9df31f7aad00",
        "Image": "nextcloud/aio-talk:latest",
        "Labels": {
            "com.centurylinklabs.watchtower.monitor-only": "true",
            "org.opencontainers.image.revision": "730b32866097ef7fc46b327aafe41ab267858e09",
            "org.opencontainers.image.source": "https://github.com/coturn/coturn",
            "org.opencontainers.image.version": "4.6.2-r0"
        },
        "OnBuild": null,
        "OpenStdin": false,
        "StdinOnce": false,
        "Tty": false,
        "User": "talk",
        "Volumes": {
            "/var/lib/coturn": {}
        },
        "WorkingDir": ""
    },
    "Created": "2023-05-02T09:01:40.698525439Z",
    "Driver": "overlay2",
    "ExecIDs": null,
    "GraphDriver": {
        "Data": {
            "LowerDir": "/var/lib/docker/overlay2/1234-init/diff:/var/lib/docker/overlay2/1234/diff:/var/lib/docker/overlay2/1234/diff:/var/lib/docker/overlay2/1234/diff:/var/lib/docker/overlay2/1234/diff:/var/lib/docker/overlay2/1234/diff:/var/lib/docker/overlay2/1234/diff:/var/lib/docker/overlay2/1234/diff:/var/lib/docker/overlay2/1234/diff:/var/lib/docker/overlay2/1234/diff:/var/lib/docker/overlay2/1234/diff:/var/lib/docker/overlay2/1234/diff",
            "MergedDir": "/var/lib/docker/overlay2/1234/merged",
            "UpperDir": "/var/lib/docker/overlay2/1234/diff",
            "WorkDir": "/var/lib/docker/overlay2/1234/work"
        },
        "Name": "overlay2"
    },
    "HostConfig": {
        "AutoRemove": false,
        "Binds": [
            "1234:/var/lib/coturn"
        ],
        "BlkioDeviceReadBps": null,
        "BlkioDeviceReadIOps": null,
        "BlkioDeviceWriteBps": null,
        "BlkioDeviceWriteIOps": null,
        "BlkioWeight": 0,
        "BlkioWeightDevice": null,
        "CapAdd": null,
        "CapDrop": null,
        "Cgroup": "",
        "CgroupParent": "",
        "CgroupnsMode": "private",
        "ConsoleSize": [
            0,
            0
        ],
        "ContainerIDFile": "",
        "CpuCount": 0,
        "CpuPercent": 0,
        "CpuPeriod": 0,
        "CpuQuota": 0,
        "CpuRealtimePeriod": 0,
        "CpuRealtimeRuntime": 0,
        "CpuShares": 0,
        "CpusetCpus": "",
        "CpusetMems": "",
        "DeviceCgroupRules": null,
        "DeviceRequests": null,
        "Devices": null,
        "Dns": null,
        "DnsOptions": null,
        "DnsSearch": null,
        "ExtraHosts": null,
        "GroupAdd": null,
        "IOMaximumBandwidth": 0,
        "IOMaximumIOps": 0,
        "IpcMode": "private",
        "Isolation": "",
        "Links": null,
        "LogConfig": {
            "Config": {},
            "Type": "json-file"
        },
        "MaskedPaths": [
            "/proc/asound",
            "/proc/acpi",
            "/proc/kcore",
            "/proc/keys",
            "/proc/latency_stats",
            "/proc/timer_list",
            "/proc/timer_stats",
            "/proc/sched_debug",
            "/proc/scsi",
            "/sys/firmware"
        ],
        "Memory": 0,
        "MemoryReservation": 0,
        "MemorySwap": 0,
        "MemorySwappiness": null,
        "NanoCpus": 0,
        "NetworkMode": "nextcloud-aio",
        "OomKillDisable": null,
        "OomScoreAdj": 0,
        "PidMode": "",
        "PidsLimit": null,
        "PortBindings": {
            "3478/tcp": [
                {
                    "HostIp": "",
                    "HostPort": "3478"
                }
            ],
            "3478/udp": [
                {
                    "HostIp": "",
                    "HostPort": "3478"
                }
            ]
        },
        "Privileged": false,
        "PublishAllPorts": false,
        "ReadonlyPaths": [
            "/proc/bus",
            "/proc/fs",
            "/proc/irq",
            "/proc/sys",
            "/proc/sysrq-trigger"
        ],
        "ReadonlyRootfs": false,
        "RestartPolicy": {
            "MaximumRetryCount": 0,
            "Name": "unless-stopped"
        },
        "Runtime": "runc",
        "SecurityOpt": null,
        "ShmSize": 67108864,
        "UTSMode": "",
        "Ulimits": [
            {
                "Hard": 200000,
                "Name": "nofile",
                "Soft": 200000
            }
        ],
        "UsernsMode": "",
        "VolumeDriver": "",
        "VolumesFrom": null
    },
    "HostnamePath": "/var/lib/docker/containers/1234/hostname",
    "HostsPath": "/var/lib/docker/containers/1234/hosts",
    "Id": "1234",
    "Image": "sha256:1234",
    "LogPath": "/var/lib/docker/containers/1234/1234-json.log",
    "MountLabel": "",
    "Mounts": [
        {
            "Destination": "/var/lib/coturn",
            "Driver": "local",
            "Mode": "z",
            "Name": "1234",
            "Propagation": "",
            "RW": true,
            "Source": "/var/lib/docker/volumes/1234/_data",
            "Type": "volume"
        }
    ],
    "Name": "/nextcloud-aio-talk",
    "NetworkSettings": {
        "Bridge": "",
        "EndpointID": "",
        "Gateway": "",
        "GlobalIPv6Address": "",
        "GlobalIPv6PrefixLen": 0,
        "HairpinMode": false,
        "IPAddress": "",
        "IPPrefixLen": 0,
        "IPv6Gateway": "",
        "LinkLocalIPv6Address": "",
        "LinkLocalIPv6PrefixLen": 0,
        "MacAddress": "",
        "Networks": {
            "nextcloud-aio": {
                "Aliases": [
                    "9df31f7aad00",
                    "e8c2bbd6b9c5"
                ],
                "DriverOpts": null,
                "EndpointID": "1234",
                "Gateway": "172.22.0.1",
                "GlobalIPv6Address": "",
                "GlobalIPv6PrefixLen": 0,
                "IPAMConfig": null,
                "IPAddress": "172.22.0.4",
                "IPPrefixLen": 16,
                "IPv6Gateway": "",
                "Links": null,
                "MacAddress": "1234",
                "NetworkID": "1234"
            }
        },
        "Ports": {
            "3478/tcp": [
                {
                    "HostIp": "0.0.0.0",
                    "HostPort": "3478"
                }
            ],
            "3478/udp": [
                {
                    "HostIp": "0.0.0.0",
                    "HostPort": "3478"
                }
            ],
            "5349/tcp": null,
            "5349/udp": null
        },
        "SandboxID": "1234",
        "SandboxKey": "/var/run/docker/netns/1234",
        "SecondaryIPAddresses": null,
        "SecondaryIPv6Addresses": null
    },
    "Path": "start.sh",
    "Platform": "linux",
    "Portainer": {
        "ResourceControl": {
            "Id": 10,
            "ResourceId": "1234",
            "SubResourceIds": [],
            "Type": 1,
            "UserAccesses": [],
            "TeamAccesses": [],
            "Public": false,
            "AdministratorsOnly": true,
            "System": false
        }
    },
    "ProcessLabel": "",
    "ResolvConfPath": "/var/lib/docker/containers/1234/resolv.conf",
    "RestartCount": 0,
    "State": {
        "Dead": false,
        "Error": "",
        "ExitCode": 0,
        "FinishedAt": "0001-01-01T00:00:00Z",
        "Health": {
            "FailingStreak": 5,
            "Log": [
                {
                    "End": "2023-05-02T11:02:41.479493313+02:00",
                    "ExitCode": -1,
                    "Output": "Health check exceeded timeout (30s)",
                    "Start": "2023-05-02T11:02:11.398403008+02:00"
                },
                {
                    "End": "2023-05-02T11:03:41.549542995+02:00",
                    "ExitCode": -1,
                    "Output": "Health check exceeded timeout (30s)",
                    "Start": "2023-05-02T11:03:11.484535658+02:00"
                },
                {
                    "End": "2023-05-02T11:04:41.645220721+02:00",
                    "ExitCode": -1,
                    "Output": "Health check exceeded timeout (30s)",
                    "Start": "2023-05-02T11:04:11.559883042+02:00"
                },
                {
                    "End": "2023-05-02T11:05:41.753046105+02:00",
                    "ExitCode": -1,
                    "Output": "Health check exceeded timeout (30s)",
                    "Start": "2023-05-02T11:05:11.650372846+02:00"
                },
                {
                    "End": "2023-05-02T11:06:41.864811584+02:00",
                    "ExitCode": -1,
                    "Output": "Health check exceeded timeout (30s)",
                    "Start": "2023-05-02T11:06:11.759649982+02:00"
                }
            ],
            "Status": "unhealthy"
        },
        "OOMKilled": false,
        "Paused": false,
        "Pid": 63500,
        "Restarting": false,
        "Running": true,
        "StartedAt": "2023-05-02T09:01:41.397090915Z",
        "Status": "running"
    }
}`

**Comment by szaimen** [comment]
Hi, the healthcheck is working fine on my end so I suspect a network infrastructure problem on your end. 

You can try to debug this your self by going into the container with `sudo docker exec -it nextcloud-aio-talk bash` and the running `set -x; (nc -z localhost 8081 && nc -z localhost 8188 && nc -z localhost 4222 && nc -z localhost "$TALK_PORT" && nc -z "$NC_DOMAIN" "$TALK_PORT") || exit 1`.

**Comment by punkyard** [comment]
thanks @szaimen 
it received: 

+ nc -z localhost 8081
+ nc -z localhost 8188
+ nc -z localhost 4222
+ nc -z localhost 3478
+ nc -z notre.rez0.net 3478

but the prompt keeps flashing and doesn't come back to :/$

**Reply by szaimen** [reply]
> * nc -z notre.rez0.net 3478

I suppose this one fails?

**Reply by szaimen** [reply]
So apparently you forgot to open port/forward 3478/udp and 3478/tcp in the firewall/router?

**Reply by Fuseteam** [reply]
@szaimen i ran into this same issue, apparently i had to add it to the reverse proxy config as follows:
```
server {
        listen 3478 ssl http2;
        server_name example.com;

        location / {
                proxy_pass http://<local-ip>:3478$request_uri;

                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Forwarded-Port $server_port;
                proxy_set_header X-Forwarded-Scheme $scheme;
                proxy_set_header X-Forwarded-Proto $scheme;
                proxy_set_header X-Real-IP $remote_addr;
                proxy_set_header Accept-Encoding "";
                proxy_set_header Host $host;

                client_body_buffer_size 512k;
                proxy_read_timeout 86400s;
                client_max_body_size 0;

                # Websocket
                proxy_http_version 1.1;
                proxy_set_header Upgrade $http_upgrade;
                proxy_set_header Connection $connection_upgrade;
        }



        ssl_certificate /etc/letsencrypt/live/team.smartsecureweb.online/fullchain.pem; # managed by Certbot
        ssl_certificate_key /etc/letsencrypt/live/team.smartsecureweb.online/privkey.pem; # managed by Certbot

        ssl_session_timeout 1d;
        ssl_session_cache shared:MozSSL:10m; # about 40000 sessions
        ssl_session_tickets off;

        ssl_protocols TLSv1.2 TLSv1.3;
        ssl_ciphers ECDHE-ECDSA-AES128-GCM-SHA256:ECDHE-RSA-AES128-GCM-SHA256:ECDHE-ECDSA-AES256-GCM-SHA384:ECDHE-RSA-AES256-GCM-SHA384:ECDHE-ECDSA-CHACHA20-POLY1305:ECDHE-RSA-CHACHA20-POLY1305:DHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES256-GCM-SHA384:DHE-RSA-CHACHA20-POLY1305;
        ssl_prefer_server_ciphers on;
}

```
Might be overkill, i based it on the example config in the readme

**Reply by Fuseteam** [reply]
looks like the container is healthy but i can't make calls hmmmm i think it may be because nginx isn't passing through udp

**Comment by punkyard** [comment]
Docker overrides the firewall. Talk has been working fine until last mastercontainer update.

**Reply by szaimen** [reply]
Did you check that the port is correctly forrwarded in your router?

**Reply by punkyard** [reply]
it worked before, I didn't think about anything yet ^^
which router? my instance is on a vps, behind cloudflare and npm

**Reply by punkyard** [reply]
I've added these two rules in ufw, and tried the command you sent me, and I get the same answer:
+ nc -z localhost 8081
+ nc -z localhost 8188
+ nc -z localhost 4222
+ nc -z localhost 3478
+ nc -z notre.rez0.net 3478

Talk container is still tagged as unhealthy

**Reply by szaimen** [reply]
> behind cloudflare

Cloudflare tunnel? So maybe cloudflare blocks this check?

**Reply by punkyard** [reply]
since when?
can I deliver some more logs from somewhere?

**Reply by szaimen** [reply]
Are you using cloudflare tunnel or proxy?

**Reply by punkyard** [reply]
tunnel for portainer
proxy for NC

**Reply by szaimen** [reply]
can you for a test disable cloudflare proxy for nc and check if that makes it work?

**Reply by punkyard** [reply]
ok, nextcloud-aio-talk is now commented as "healthy" in portainer after I disabled cloudflare proxy and rebuilt the container.

**Reply by szaimen** [reply]
cc @Zoey2936 FYI

**Reply by punkyard** [reply]
Thanks for the lead then @szaimen!
Now, is there a howto to fix since the last update, and keep proxy protection?

**Reply by szaimen** [reply]
I fear this is yet another limitation when using Cloudflare Proxy. I added it to the readme: https://github.com/nextcloud/all-in-one/commit/d58bb56dfa21c1c6d44612629d487a05c5b3f212

**Reply by punkyard** [reply]
that's very much welcome. Have a good rest of the day ✌️

**Comment by punkyard** [comment]
let me ask you a question:
I came to AiO because NC was unstable, filled with logs, DB breakdowns. AiO is clearly the best solution according to people's advice and my short experience. But can you tell me why I still get long logs in the Admin / Loggin?
Everything isn't supposed to work together inside containers, without me doing any mistake?
I can't understand that.. can you?

Thanks any how for the great work you're doing. So far, I'm still testing NC AiO but I do hope volunteer projects will enter this instance, as soon as it's 'under control' ^^

**Reply by szaimen** [reply]
> But can you tell me why I still get long logs in the Admin / Loggin?

what kind of logs do you see there?

**Reply by punkyard** [reply]
So many different ones, like:

Error: Redis::get(): php_network_getaddresses: getaddrinfo failed: Name does not resolve at /var/www/html/lib/private/Memcache/Redis.php#48

OCA\Collectives\Service\NotPermittedException: FederatedUser is not complete

At the moment it's full with errors of connection with my email server which is down.
But I'll message more of them when I've fixed this. Btw, which email server in portainer would you recommand? I'm trying to install mailcow at the moment, which I couldn't make run for now.

**Reply by szaimen** [reply]
> Error: Redis::get(): php_network_getaddresses: getaddrinfo failed: Name does not resolve at /var/www/html/lib/private/Memcache/Redis.php#48

This might happen when your server is overloaded or when shuttiing down containers.

> OCA\Collectives\Service\NotPermittedException: FederatedUser is not complete

Thi might be a problem of the collectives app. Feel free to report an error to https://github.com/nextcloud/collectives/issues

> At the moment it's full with errors of connection with my email server which is down.

Unrelated to AIO indeed

**Comment by gokou340** [comment]
I'm having this exact same issue, but cloudflare proxy is turned off.

EDIT: Running the following command exits me back to prompt: set -x; (nc -z localhost 8081 && nc -z localhost 8188 && nc -z localhost 4222 && nc -z localhost "$TALK_PORT" && nc -z "$NC_DOMAIN" "$TALK_PORT") || exit 1

I am using Nginx Proxy Manager as a reverse proxy.

**Reply by Anexgohan** [reply]
same issue as above, all containers working fine, except nextcloud-aio-talk is unhealthy, on docker behind nginx proxy manager.
I can send text and image, but video calls are not happening.

Starting part of the log:
```
0: (17): INFO: Black listing: 10.0.0.0-10.255.255.255
0: (17): INFO: Black listing: 100.64.0.0-100.127.255.255
0: (17): INFO: Black listing: 127.0.0.0-127.255.255.255
0: (17): INFO: Black listing: 169.254.0.0-169.254.255.255
0: (17): INFO: Black listing: 172.16.0.0-172.31.255.255
0: (17): INFO: Black listing: 192.0.0.0-192.0.0.255
0: (17): INFO: Black listing: 192.0.2.0-192.0.2.255
0: (17): INFO: Black listing: 192.88.99.0-192.88.99.255
0: (17): INFO: Black listing: 192.168.0.0-192.168.255.255
0: (17): INFO: Black listing: 198.18.0.0-198.19.255.255
0: (17): INFO: Black listing: 198.51.100.0-198.51.100.255
0: (17): INFO: Black listing: 203.0.113.0-203.0.113.255
0: (17): INFO: Black listing: 240.0.0.0-255.255.255.255
0: (17): INFO: Coturn Version Coturn-4.6.2 'Gorst'
0: (17): INFO: Max number of open files/sockets allowed for this process: 200000
0: (17): INFO: Due to the open files/sockets limitation, max supported number of TURN Sessions possible is: 100000 (approximately)
0: (17): INFO: 

==== Show him the instruments, Practical Frost: ====

0: (17): INFO: OpenSSL compile-time version: OpenSSL 3.1.1 30 May 2023 (0x30100010)
0: (17): INFO: TLS 1.3 supported
0: (17): INFO: DTLS 1.2 supported
0: (17): INFO: TURN/STUN ALPN supported
0: (17): INFO: Third-party authorization (oAuth) supported
0: (17): INFO: GCM (AEAD) supported
0: (17): INFO: SQLite supported, default database location is /var/lib/coturn/turndb
0: (17): INFO: Redis supported
0: (17): INFO: PostgreSQL supported
0: (17): INFO: MySQL supported
0: (17): INFO: MongoDB supported
0: (17): INFO: Default Net Engine version: 3 (UDP thread per CPU core)
0: (17): INFO: Domain name: 
0: (17): INFO: Default realm: nextcloud.test.com
0: (17): ERROR: CONFIG: Empty cli-password, and so telnet cli interface is disabled! Please set a non empty cli-password!
0: (17): INFO: Certificate file found: //turn_server_cert.pem
0: (17): INFO: Private key file found: //turn_server_pkey.pem
0: (17): WARNING: NO EXPLICIT LISTENER ADDRESS(ES) ARE CONFIGURED
0: (17): INFO: ===========Discovering listener addresses: =========
0: (17): INFO: Listener address to use: 127.0.0.1
0: (17): INFO: Listener address to use: 172.20.0.4
0: (17): INFO: =====================================================
0: (17): INFO: Total: 1 'real' addresses discovered
0: (17): INFO: =====================================================
0: (17): WARNING: NO EXPLICIT RELAY ADDRESS(ES) ARE CONFIGURED
0: (17): INFO: ===========Discovering relay addresses: =============
0: (17): INFO: Relay address to use: 172.20.0.4
0: (17): INFO: =====================================================
0: (17): INFO: Total: 1 relay addresses discovered
0: (17): INFO: =====================================================
0: (17): WARNING: Cannot create pid file: /var/tmp/turnserver.pid
0: (17): INFO: pid file created: /tmp/turnserver.pid
Logger plugins folder: /usr/local/lib/janus/loggers
---------------------------------------------------
  Starting Meetecho Janus (WebRTC Server) v0.13.4
---------------------------------------------------

Checking command line arguments...
Debug/log level is 3
Debug/log timestamps are disabled
Debug/log colors are disabled
[WARN] Janus is deployed on a private address (172.20.0.4) but you didn't specify any STUN server! Expect trouble if this is supposed to work over the internet and not just in a LAN...
[WARN] IPv6 disabled, will only create VideoRoom forwarders to IPv4 addresses
[WARN] libcurl not available, Streaming plugin will not have RTSP support
[WARN] libogg not available, Streaming plugin will not have file-based Opus streaming
[WARN] No Unix Sockets server started, giving up...
[WARN] The 'janus.transport.pfunix' plugin could not be initialized
natsclient.go:108: Connection established to nats://127.0.0.1:4222 (NAQTTVY6KOCXAJAEFRTHGYDHDSGQZS7657237M5IGKCDUOFSWCTRH7D227Y6U)
grpc_common.go:167: WARNING: No GRPC server certificate and/or key configured, running unencrypted
grpc_common.go:169: WARNING: No GRPC CA configured, expecting unencrypted connections
backend_storage_static.go:72: Backend backend-1 added for https://nextcloud.test.com/
hub.go:200: Using a maximum of 8 concurrent backend connections per host
hub.go:207: Using a timeout of 10s for backend connections
hub.go:303: Not using GeoIP database
mcu_janus.go:294: Connected to Janus WebRTC Server 0.13.4 by Meetecho s.r.l.
mcu_janus.go:300: Found JANUS VideoRoom plugin 0.0.9 by Meetecho s.r.l.
mcu_janus.go:305: Data channels are supported
mcu_janus.go:309: Full-Trickle is enabled
mcu_janus.go:311: Maximum bandwidth 1048576 bits/sec per publishing stream
mcu_janus.go:312: Maximum bandwidth 2097152 bits/sec per screensharing stream
mcu_janus.go:318: Created Janus session 6914369362571492
mcu_janus.go:325: Created Janus handle 7729920353157016
main.go:263: Using janus MCU
hub.go:385: Using a timeout of 10s for MCU requests
backend_server.go:111: No IPs configured for the stats endpoint, only allowing access from 127.0.0.1
main.go:339: Listening on 0.0.0.0:8081
```

**Comment by vrjula** [comment]
Similar issue here thus far.  My setup:

Firewall: pfSense
Reverse proxy: HAProxy, running in pfSense
Container: Nextcloud AIO, running in a Proxmox LXC

I'm running a local instance, without exposure to the cloud.  I'm forwarding port 3478 from the frontend to the backend.  Running command from @szaimen above within the container does not error out, so I know the URL is resolving within the container.  But the turnserver check on the Admin Talk GUI does not pass when I use the URL:3478.  It does pass when I use [local-ip-address]:3478.

Functionally, when I try to connect with video, it just keeps connecting then disconnecting.  Is there a specific log somewhere I can watch to see why it keeps disconnecting?

**Reply by dsaderholm** [reply]
Did you ever solve this?

**Reply by Fuseteam** [reply]
you need to port forward 3478 in your pfsense directly to the aio container (by passing the proxy)
the issue is that 3478 uses udp traffic, if your reverse proxy only passes tcp traffic it won't work

**Comment by dsaderholm** [comment]
I am also having the same issue. I use Nginx Proxy Manager and Cloudflare, but with proxy off. What am I doing wrong?

**Comment by szaimen** [comment, marked as answer]
Hi, see https://github.com/nextcloud/all-in-one/discussions/4592

**Comment by iamtherobin** [comment]
Want to point people's attention to my post in the answered ticket: https://github.com/nextcloud/all-in-one/discussions/4592#discussioncomment-9680503

My finding is similar to @Fuseteam in the sense the only way I could clear the unhealthy status was to add a TCP proxy into my reverse proxy. However, that only clears unhealthy status and won't make necessary make Talk work since it needs UDP as well. I describe my DNS setup (dual stack compatible) to make Talk work and clear the unhealthy status in the post.

**Reply by Fuseteam** [reply]
I do suspect that if your reverse proxy can forward UDP traffic, it should just work™
